### PR TITLE
Clarify how attributions are made

### PR DESCRIPTION
**Pull request by sdrave**
As discussed in #2218.

Turns out, we already state that attributions are made unless the author does not want it. I have added some sentences clarifying that

- Attributions mention GitHub user name and name from GitHub profile in the future
- `AUTHORS.md` is updated as part of the release process.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -60,10 +60,11 @@ of your code under the same license.
 ## Attribution
 
 When you have contributed code to pyMOR you and the content of your
```
Comment (pmli) [inline CONTRIBUTING.md:62]:
```suggestion
When you have contributed code to pyMOR, you and the content of your
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -60,10 +60,11 @@ of your code under the same license.
...
-contribution will be mentioned in the project's AUTHORS.md file.
-Contributions are grouped by release, so if you have contributed
-code to multiple releases, you will be mentioned for each of
-these releases. Moreover, you may add attribution notices (e.g.
+contribution will be mentioned in the project's AUTHORS.md file
+along with your GitHub user name and the name you have set in your
+GitHub Profile. Contributions are grouped by release, so if you have
```
Comment (pmli) [inline CONTRIBUTING.md:65]:
```suggestion
GitHub profile. Contributions are grouped by release, so if you have
```

File: docs/RELEASE_PROCESS.md [hunk 3]
```diff
@@ -46,6 +46,8 @@ The release manager is responsible for ensuring that the following steps are per
         (new feature PRs have to wait until after the release to get merged).
   - [ ] Determine number of commits and so on for the release notes and add these information.
         Merge the release notes branch into `main` when they are finished.
+  - [ ] Update `AUTHORS.md` to include attributions for relevant contributions made to the
+        relase by non-main devs.
```
Comment (pmli) [inline docs/RELEASE_PROCESS.md:50]:
```suggestion
        release by non-main devs.
```

**Review by pmli — COMMENTED** [review]
Looks good to me, just minor comments.

For consistency, I would prefer to update all older contributions to use GitHub user names, but I could live with it as is.

**Comment by drittelhacker** [comment]
+1 for @pmli's consistency argument

**Comment by sdrave** [comment]
I have replaced email addresses by GitHub user names where possible.

**Review by pmli — APPROVED** [review]

**Pull request merged**